### PR TITLE
Enforce console logging for peon process

### PR DESCRIPTION
**Pull request by FrankChen021**
### Description

Fixes the problem mentioned in #10147 by @clintropolis that peon process's log is output to middle manager's log file.

Currently all Druid processes share the same log4j2 configuration file located in `_common` directory. Since peon processes are spawned by middle manager process, they derivate the environment variables from the middle manager. These variables include those in the `log4j2.xml` controlling to which file the logger writes the log.

But current task logging mechanism requires the peon processes to output the log to console so that the middle manager can redirect the console output to a file and upload this file to task log storage.

So, this PR imposes this requirement to peon processes, **whatever the configuration is in the shared log4j2.xml, peon processes always write the log to console**.

<hr>

##### Key changed/added classes in this PR
 * added `ConsoleLoggingEnforcementConfigurationFactory.java`, to enfore `ConsoleAppender` is used
 * updated `ForkingTaskRunner.java`, to pass a `-Dlog4j2.configurationFactory` argument to peon

<hr>

After applying this patch, we can see from the middle manager's log that under file appender configuration, the task long won't output to the middle manager's log file, and the middle manager starts the peon with the newly added jvm parameter `-Dlog4j2.configurationFactory`.

```text
2021-12-18T08:58:49,726 INFO [WorkerTaskManager-NoticeHandler] org.apache.druid.indexing.worker.WorkerTaskManager - Task[index_parallel_wikipedia_klcfmdop_2021-12-18T08:58:49.704Z] started.
2021-12-18T08:58:49,727 INFO [forking-task-runner-1] org.apache.druid.indexing.overlord.ForkingTaskRunner - Running command: java -cp ... -Dlog4j2.configurationFactory=org.apache.druid.indexing.common.tasklogs.ConsoleLoggingEnforcementConfigurationFactory org.apache.druid.cli.Main internal peon var/druid/task/index_parallel_wikipedia_klcfmdop_2021-12-18T08:58:49.704Z/task.json var/druid/task/index_parallel_wikipedia_klcfmdop_2021-12-18T08:58:49.704Z/ff580551-d0f6-4da7-bdb9-c88cdc110d76/status.json var/druid/task/index_parallel_wikipedia_klcfmdop_2021-12-18T08:58:49.704Z/ff580551-d0f6-4da7-bdb9-c88cdc110d76/report.json
2021-12-18T08:58:49,736 INFO [forking-task-runner-1] org.apache.druid.indexing.overlord.ForkingTaskRunner - Logging task index_parallel_wikipedia_klcfmdop_2021-12-18T08:58:49.704Z output to: var/druid/task/index_parallel_wikipedia_klcfmdop_2021-12-18T08:58:49.704Z/log
2021-12-18T08:58:59,417 INFO [forking-task-runner-1] org.apache.druid.indexing.common.tasklogs.FileTaskLogs - Wrote task log to: var/druid/indexing-logs/index_parallel_wikipedia_klcfmdop_2021-12-18T08:58:49.704Z.log
2021-12-18T08:58:59,418 INFO [forking-task-runner-1] org.apache.druid.indexing.common.tasklogs.FileTaskLogs - Wrote task report to: var/druid/indexing-logs/index_parallel_wikipedia_klcfmdop_2021-12-18T08:58:49.704Z.report.json
2021-12-18T08:58:59,418 INFO [forking-task-runner-1] org.apache.druid.indexing.overlord.ForkingTaskRunner - Process exited successfully for task: index_parallel_wikipedia_klcfmdop_2021-12-18T08:58:49.704Z
2021-12-18T08:58:59,424 INFO [forking-task-runner-1] org.apache.druid.indexing.overlord.ForkingTaskRunner - Removing task directory: var/druid/task/index_parallel_wikipedia_klcfmdop_2021-12-18T08:58:49.704Z
2021-12-18T08:58:59,427 INFO [WorkerTaskManager-NoticeHandler] org.apache.druid.indexing.worker.WorkerTaskManager - Task [index_parallel_wikipedia_klcfmdop_2021-12-18T08:58:49.704Z] completed with status [SUCCESS].
```

This PR has:
- [X] been self-reviewed.
   - [ ] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [X] added documentation for new or modified features or behaviors.
- [X] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/dev/license.md)
- [X] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [x] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [ ] added integration tests.
- [X] been tested in a test Druid cluster.

### CODE REVIEW

**Comment by FrankChen021** [comment]
Marked as WIP to add some UT tests later, but the implementation seems like no big change.

**Comment by FrankChen021** [comment]
Hi @jihoonson @clintropolis  this PR should be merged before we start a new release. Please take a look at it at you convenience. Thank you.

File: docs/configuration/logging.md [hunk 1]
```diff
@@ -23,9 +23,12 @@ title: "Logging"
   -->
 
 
-Apache Druid processes will emit logs that are useful for debugging to the console. Druid processes also emit periodic metrics about their state. For more about metrics, see [Configuration](../configuration/index.md#enabling-metrics). Metric logs are printed to the console by default, and can be disabled with `-Ddruid.emitter.logging.logLevel=debug`.
+Apache Druid processes will emit logs that are useful for debugging to log files. 
+These processes also emit periodic [metrics](../configuration/index.md#enabling-metrics) about their state.
+Metric logs are can be disabled with `-Ddruid.emitter.logging.logLevel=debug`.
```
Comment (kfaraz) [inline docs/configuration/logging.md:28]:
```suggestion
Metric info logs can be disabled with `-Ddruid.emitter.logging.logLevel=debug`.
```

**Review by kfaraz — APPROVED** [review]
Thanks for these changes, @FrankChen021 .
Added minor nitpicks. Otherwise LGTM.

**Comment by FrankChen021** [comment]
Thank you @kfaraz for the review. @abhishekagarwal87 This PR should be backport to the 0.23 branch.

**Comment by abhishekagarwal87** [comment]
@FrankChen021 - can you create a backport PR?

**Comment by FrankChen021** [comment]
@abhishekagarwal87 #12527 is created

**Pull request merged**